pkg/security/admission/admission_test.go

```diff
@@ -553,38 +572,42 @@ func TestAssignSecurityContext(t *testing.T) {
 		},
 	}
 
-	for k, v := range testCases {
-		errs := assignSecurityContext(provider, v.pod, nil)
-		if v.shouldValidate && len(errs) > 0 {
-			t.Errorf("%s expected to validate but received errors %v", k, errs)
-			continue
-		}
-		if !v.shouldValidate && len(errs) == 0 {
-			t.Errorf("%s expected validation errors but received none", k)
-			continue
-		}
+	for i := 0; i < 2; i++ {
```

Comment:
There doesn't have to be one?  Just running twice and swapping the init containers and containers ensures that we test both paths and they fail identically.